mypy/mixedtraverser.py

```diff
@@ -0,0 +1,89 @@
+from typing import Optional
+
+from mypy.nodes import (
+    MypyFile, Var, FuncItem, ClassDef, AssignmentStmt, OperatorAssignmentStmt, ForStmt, WithStmt,
+    CastExpr, TypeApplication, TypeAliasExpr, TypeVarExpr, TypedDictExpr, NamedTupleExpr,
+    EnumCallExpr, PromoteExpr, NewTypeExpr
+)
+from mypy.types import Type
+from mypy.traverser import TraverserVisitor
+from mypy.typetraverser import TypeTraverserVisitor
+
+
+class MixedTraverserVisitor(TraverserVisitor, TypeTraverserVisitor):
+    """Recursive traversal of both Node and Type objects."""
+
+    # Symbol nodes
+
+    def visit_var(self, var: Var) -> None:
+        self.visit_optional_type(var.type)
+
+    def visit_func(self, o: FuncItem) -> None:
+        super().visit_func(o)
+        self.visit_optional_type(o.type)
+
+    def visit_class_def(self, o: ClassDef) -> None:
+        super().visit_class_def(o)
+        info = o.info
+        if info:
```

Comment:
I think there is something missing here. For various auto-generated `TypeInfo`s there are methods/variables without corresponding syntactic nodes. These will be never reached unless you actually loop through the namespace. We used to have a bunch of crashes because of this.

Reply:
What sorts of crashes? Since there are generated in code, we can perhaps assume that they have valid type arguments. I'll add a comment here.

Reply:
Crashes appeared in something related to this:
```python
class NT(NamedTuple):
    x: List[int, str]
```
Anyway, we have very good test coverage for these, so if there any problems, they will be likely caught when we will start enabling the tests.